### PR TITLE
MGMT-12697: Reject unknown json fields

### PR DESCRIPTION
**Pull request by jhernand**
Currently the API accepts requests with unknown fields and silently ignores them. This patch changes it so that it rejects those requests. For example a request like this:

```
POST /api/assisted-install/v2/clusters/a67d9ab8-ef3f-4176-ac69-337bb26ae135 <<.
{
  "base_dns_doman": "home.arpa"
}
```

Will result in the following response:

```
HTTP/1.1 400 Bad Request
{
  "code": 400,
  "message": "parsing clusterUpdateParams body from \"\" failed, because json: unknown field \"base_dns_doman\""
}
```

The error message isn't very nice, but similar to error messages generated by other wrong JSON documents. Should probably be improved in the future.

This feature is enabled by default, but it can be disabled setting the `ENABLE_REJECT_UNKNOWN_FIELDS` environment variable to `false`.

Related: https://issues.redhat.com/browse/MGMT-12697

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by jhernand** [comment]
Note that this pull request includes two patches:

- The first upgrades the version of go-swagger that we use. It uses a version that has't been released yet and it is very long, but it only contains minor changes. Once you have reviewed a few files it is not worth reviewing the rest.

- The second commit is the one that introduces the real feature, please focus your review on that one.

**Comment by rccrdpccl** [comment]
Are we going to add a test about this somewhere?

**Comment by jhernand** [comment]
> Are we going to add a test about this somewhere?

Yes, in the next version of the pull request.

File: Dockerfile.assisted-service-build [hunk 1]
```diff
@@ -11,7 +11,22 @@ ENV TOOLS=/tools/
 ENV PATH="$VIRTUAL_ENV/bin:$GOROOT/bin:$GOPATH/bin:$TOOLS:$PATH"
 
 COPY --from=quay.io/app-sre/golangci-lint:v1.46.0 /usr/bin/golangci-lint /usr/bin/golangci-lint
-COPY --from=quay.io/goswagger/swagger:v0.28.0 /usr/bin/swagger /usr/bin/goswagger
```
Comment (osherdp) [inline Dockerfile.assisted-service-build:14]: you need to update here as well: https://github.com/openshift/assisted-service/blob/57ed23bb318c29daf89bde6db3e77257723574bc/ci-images/Dockerfile.code-generation#L6

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
> @osherdp: GitHub didn't allow me to request PR reviews from the following users: be, better, for, I, change, you, should, a, reviewer, this, think.
> 
> Note that only [openshift members](https://github.com/orgs/openshift/people) and repo collaborators can review this PR, and authors cannot review their own PRs.

:man_facepalming:

File: Dockerfile.assisted-service-build (hunk 1 above)
Comment (jhernand) [reply Dockerfile.assisted-service-build:14]: Thanks @osherdp, done.

File: subsystem/low_level_api_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/go-openapi/swag"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/assisted-service/models"
+)
+
+var _ = Describe("Low level API behaviours", func() {
+	It("Rejects unknown JSON fields", func() {
+		// Create a request and marshal it:
```
Comment (jhernand) [inline subsystem/low_level_api_test.go:19]: @rccrdpccl here is the test that you requested.

**Pull request merged**